src/visualizers/panels/ExecutionIndex/ExecutionIndexControl.js

```diff
@@ -38,45 +41,88 @@ define([
+        let removedExecutions = beforeClickIds.filter(id => !afterClickIds.includes(id));
+        let added = addedExecutions.length > 0;
+        let updatedExecutions = added ? addedExecutions : removedExecutions;
+        this._updateGraphData(...updatedExecutions, added);
+    };
 
-        this._logger.info(`setting execution ${id} to ${bool ? 'displayed' : 'hidden'}`);
+    ExecutionIndexControl.prototype._updateGraphData = function (id, bool) {
         this.displayedExecutions[id] = bool;
-
-        // If we just crossed the multi line threshold, then update all the lines
-        isMultiLine = this.displayedExecCount() > 1;
-        if (isMultiLine !== wasMultiLine) {
-            // Refresh the other lines visible
-            otherLines = Object.keys(this.displayedExecutions)
-                .filter(eId => this.displayedExecutions[eId] && (eId !== id))
-                .map(id => this._linesForExecution[id] || [])
-                .reduce((l1, l2) => l1.concat(l2), []);
-
-            this._updateLines(otherLines, false);
-            this._updateLines(otherLines, true);
+        let displayedGraphExecIds = Object.keys(this.displayedExecutions)
+            .filter((id) => !!this.displayedExecutions[id]);
+        if (this._currentPlotsDataId) {
+            this._widget.removeNode(this._currentPlotsDataId);
+        }
+        if ( this.displayedExecCount() > 0) {
+            this._consolidateGraphData(displayedGraphExecIds);
         }
-
-        this._updateLines(lines, bool);
     };
 
-    ExecutionIndexControl.prototype._updateLines = function (lines, added) {
-        var action = added ? 'addNode' : 'removeNode';
+    ExecutionIndexControl.prototype._consolidateGraphData = function (graphExecIDs) {
+        let graphIds = graphExecIDs.map(execId => this._graphsForExecution[execId]);
+        let graphDescs = graphIds.map(id => this._getObjectDescriptor(id)).filter(desc => !!desc);
+        if(graphDescs.length > 0){
+            let consolidatedDesc = this._combineGraphDesc(graphDescs);
+            this._currentPlotsDataId = consolidatedDesc.id;
+            let plotlyJSON = this._plotlyDescExtractor.descToPlotlyJSON(consolidatedDesc);
+            plotlyJSON.type = 'graph';
+            this._widget.addNode(plotlyJSON);
+        }
+    };
 
-        // If removing, just get the ids
-        lines = !added ? lines : lines.map(line => this._getObjectDescriptor(line))
-            .filter(line => !!line);
+    ExecutionIndexControl.prototype._combineGraphDesc = function (graphDescs) {
+        const isMultiGraph = this.displayedExecCount() > 1;
+        if (!isMultiGraph) {
+            return graphDescs[0];
+        } else {
+            let consolidatedDesc = null;
+
+            graphDescs.forEach((desc) => {
+                if (!consolidatedDesc) {
+                    consolidatedDesc = JSON.parse(JSON.stringify(desc));
+                    consolidatedDesc.title = consolidatedDesc.title || `Graph`;
```

Comment:
Since a default value is defined both here and on line 95, it might be nice to just define a method like `getDisplayName(desc)` which contains the default title (defined in only one place). It could also append the abbreviation.

Reply:
A function called `setDisplayTitle` has been added with a optional boolean flag to include `abbr` which is the execution Abbrevation for this graph/subgraph.